vignettes/customization.Rmd

```diff
@@ -0,0 +1,222 @@
+How to have a faster workflow?
+
+* Use a simpler package for previewing your style. This might be over the top for a single package but if you are working on a template package, the template package can contain minimal vignettes, reference pages for testing!
+* Do not rebuild your package every time: e.g. only re-build the home page via `build_home()` to get a preview of the home sidebar, or of the footer and navbar.
+* If you edit CSS, you can run `init_site()` to have them copied to your website folder, and then refresh the website in your browser.
+
+## More assets
+
+If you need to store more CSS rules, some JS code, an SVG used by your CSS, etc.
+
+* Store the files under `inst/pkgdown/assets/BS4` (if you're using Bootstrap 4) e.g. `inst/pkgdown/assets/BS4/cool-styling.css`.
+* They will be copied to the root of your website folder by pkgdown (by `init_site()`).
+* For JS and CSS files, reference them in a `inst/pkgdown/after-head.html` template (see the section "Other layout changes"), e.g.
+
+```html
+<link href="{{#site}}{{root}}{{/site}}cool-styling.css" rel="stylesheet">
+```
+
+## Layout
+
+You will want to alter the layout of your website if you are trying to e.g.
+
+* customize elements of the navbar, footer, home sidebar,
+* add a few HTML lines corresponding to your HTML provider,
+* change the information provided in e.g. the authors page.
+
+### Navbar, footer, home sidebar
+
+The navbar, footer and home sidebar can be tweaked via the pkgdown configuration file of your website.
+
+If you want to alter the elements of the navbar, refer to `?build_site` docs.
+If you want to alter the elements of the footer, refer to `?build_site` docs.
+If you want to alter the elements of the home sidebar, refer to `?build_home` docs.
+
+### Other layouts changes
+
+Other changes of layouts require tinkering with pkgdown _templates_.
+
+pkgdown uses the following whisker templates for Bootstrap 4: 
+
+```{r, results='asis', echo=FALSE}
+cat(paste0("* ", fs::path_file(fs::dir_ls(pkgdown:::path_pkgdown("templates", "BS4"))), "\n"))
+```
+
+You can override the default templates (cf `?build_site`).
+For any template for which you have **not** provided a replacement, pkgdown will use the default template (corresponding to the right Bootstrap version) so you only need to provide pkgdown with the templates you _have_ tweaked.
+
+Three of pkgdown templates are meant to be customized:
+
+* "in-header.html" inserted right before closing the head tag, 
+* "before-body.html" inserted right after opening the body tag,
+* "after-body.html" inserted right before closing the body tag.
+
+Overriding other templates is recommended for advanced users only.
+Furthermore, maintaining e.g. your own "head.html" template requires more work as you might need to keep it up-to-date with upstream changes in pkgdown itself whereas "in-header.html", "before-body.html" and "after-body.html" are by default empty.
```

Comment:
I'd make this even stronger — If you've used `head.html` in the past, we now recommend switching to ...